packages/plugin-ext/src/hosted/node/scanners/plugin-uri-factory.ts

```diff
@@ -0,0 +1,25 @@
+
+/********************************************************************************
+ * Copyright (C) 2020 Red Hat, Inc.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import URI from '@theia/core/lib/common/uri';
+import { PluginPackage } from '../../../common';
+
+export const PluginUriFactory = Symbol('PluginUriFactory');
+
+export interface PluginUriFactory {
+    createUri(pkg: PluginPackage, pkgRelativePath?: string): URI;
```

Comment:
Could you comment on how the generated URI should relate to the plugin? URI are so generic that I fail to see the exact purpose of this one. Not saying that there's not, just that it's not obvious to me.

(A URI is an address to a resource, but in itself it doesn't describe what kind of resource it is...)

Reply:
Just a comment on the function explaining what should be returned would help.

Reply:
The rationale for having the factory is in the PR description, But you are right: this kind of info belongs into the code. Thx for keeping me honest.